### ISSUE TITLE
StreamingTextResponse(stream) doesn't stream with AZURE OPEN AI 

### ISSUE THREAD

**Issue by IsuruCreative** [vercel/ai#945]
### Description

Hi, I am trying to implement basic RAG chat bot using AZURE OPEN AI, and @azure/openai SDK. But for some reason my chat response doesn't get stream into the UI. Instead, the response come as a one block of content response. Below is my chat api route.

```
import { OpenAIClient, AzureKeyCredential } from "@azure/openai";
import { OpenAIStream, StreamingTextResponse } from "ai";
import { SearchClient } from "@azure/search-documents";

const AOAI_KEY = process.env.AOAI_KEY;
const AOAI_ENDPOINT = process.env.AOAI_ENDPOINT;
const AAS_KEY = process.env.AAS_KEY;

const openai = new OpenAIClient(
  AOAI_ENDPOINT,
  new AzureKeyCredential(AOAI_KEY)
);

const search = new SearchClient(
  "https://chat-101-search.search.windows.net",
  "vector-1705422986077",
  new AzureKeyCredential(AAS_KEY)
);

// Export a POST function
export async function POST(req: Request) {
  // Get the messages from the request body
  const request = await req.json();
  const deploymentId = "chat";

  //message ques size
  const msg_length = request.messages.length;
  const user_msg = request.messages[msg_length - 1];

  let info = "";

  const msg_embeddings = await openai.getEmbeddings(
    "embedding",
    user_msg.content
  );

  // Pass the array as the value of the vector parameter
  const response = await search.search(user_msg.content, {
    select: ["chunk", "title"],
    queryLanguage: "en-us",
    vector: {
      fields: ["vector"],
      kNearestNeighborsCount: 3,
      value: msg_embeddings.data[0].embedding,
    },
    top: 3,
  });

  for await (const result of response.results) {
    // These results are the nearest neighbors to the query vector
    const title = result.document.title;
    const text = result.document.chunk;

    console.log(result.score);

    const item = `- **${title}**: ${text}\n`;
    info += item;
  }

  //prompts
  const messages = [
    {
      role: "system",
      content:
        "You are a helpful assistant. You will talk like a helpful assistant to provide solutions to user questions.",
    },
    {
      role: "user",
      content: `this is the user's question ${user_msg.content} and here is the supported content to answer the user's question  ${info}. 
    if you can't build the answer from the supported content say you can't give and answer what user ask. `,
    },
    {
      role: "assistant",
      content: "Assist user's in polite way",
    },
  ];

  const events = await openai.streamChatCompletions(deploymentId, messages, {
    maxTokens: 200,
  });

  // Respond with the stream
  const stream = OpenAIStream(events, {
    onStart: async () => {
      // This callback is called when the stream starts
      // You can use this to save the prompt to your database
    },
    onToken: async (token: string) => {
      // This callback is called for each token in the stream
      // You can use this to debug the stream or save the tokens to your database
      // console.log("token " + token);
    },
    onCompletion: async (completion: string) => {
      // This callback is called when the stream completes
      // You can use this to save the final completion to your database
      // console.log("completion " + completion);
    },
  });

  let res = new StreamingTextResponse(stream);
  return res;
}
```

and here's my Chat UI 

```
"use-client";
import clsx from "clsx";
import React, { ChangeEvent } from "react";
import { useChat } from "ai/react";
import { Avatar } from "../ui/avatar";
import { AvatarFallback, AvatarImage } from "@radix-ui/react-avatar";
import { Button } from "../ui/button";
import { useSession } from "next-auth/react";

const Chat = () => {
  const { messages, input, handleInputChange, handleSubmit, stop, isLoading } =
    useChat();

  const { data: session } = useSession();

  const {
    user: { name, email, image },
  } = session;

  console.log(messages)

  return (
    <div className="flex flex-col h-[80vh]">
      <div className="overflow-y-auto">
        <div>
          {messages.map((m) => (
            <div
              className={clsx("p-8 flex justify-start", {
                "bg-gray-200 text-gray-800": m.role === "user",
                "bg-gray-100 text-gray-800": m.role !== "user",
              })}
              key={m.id}
            >
              <div className="max-w-[1024px] flex gap-4">
                <Avatar>
                  <AvatarImage
                    src={
                      m.role === "user"
                        ? image
                        : "https://github.com/shadcn.png"
                    }
                    alt="user"
                  />
                  <AvatarFallback>IA</AvatarFallback>
                </Avatar>
                <div className="mt-2">
                  {isLoading && m.role !== "user" ? "Loading" : m.content}
                </div>
              </div>
            </div>
          ))}
        </div>
      </div>

      <div className="mt-auto flex">
        {/* <ChatInput /> */}
        <form className="w-full flex gap-2 m-4" onSubmit={handleSubmit}>
          <input
            className="rounded border w-full border-gray-300 p-2 shadow-xl"
            value={input}
            onChange={handleInputChange}
          />
          <Button onClick={stop}>Stop</Button>
          <Button type="submit">Send</Button>
        </form>
      </div>
    </div>
  );
};

export default Chat;

```

### Code example

```
import { OpenAIClient, AzureKeyCredential } from "@azure/openai";
import { OpenAIStream, StreamingTextResponse } from "ai";
import { SearchClient } from "@azure/search-documents";

const AOAI_KEY = process.env.AOAI_KEY;
const AOAI_ENDPOINT = process.env.AOAI_ENDPOINT;
const AAS_KEY = process.env.AAS_KEY;

const openai = new OpenAIClient(
  AOAI_ENDPOINT,
  new AzureKeyCredential(AOAI_KEY)
);

const search = new SearchClient(
  "https://chat-101-search.search.windows.net",
  "vector-1705422986077",
  new AzureKeyCredential(AAS_KEY)
);

// Export a POST function
export async function POST(req: Request) {
  // Get the messages from the request body
  const request = await req.json();
  const deploymentId = "chat";

  //message ques size
  const msg_length = request.messages.length;
  const user_msg = request.messages[msg_length - 1];

  let info = "";

  const msg_embeddings = await openai.getEmbeddings(
    "embedding",
    user_msg.content
  );

  // Pass the array as the value of the vector parameter
  const response = await search.search(user_msg.content, {
    select: ["chunk", "title"],
    queryLanguage: "en-us",
    vector: {
      fields: ["vector"],
      kNearestNeighborsCount: 3,
      value: msg_embeddings.data[0].embedding,
    },
    top: 3,
  });

  for await (const result of response.results) {
    // These results are the nearest neighbors to the query vector
    const title = result.document.title;
    const text = result.document.chunk;

    console.log(result.score);

    const item = `- **${title}**: ${text}\n`;
    info += item;
  }

  //prompts
  const messages = [
    {
      role: "system",
      content:
        "You are a helpful assistant. You will talk like a helpful assistant to provide solutions to user questions.",
    },
    {
      role: "user",
      content: `this is the user's question ${user_msg.content} and here is the supported content to answer the user's question  ${info}. 
    if you can't build the answer from the supported content say you can't give and answer what user ask. `,
    },
    {
      role: "assistant",
      content: "Assist user's in polite way",
    },
  ];

  const events = await openai.streamChatCompletions(deploymentId, messages, {
    maxTokens: 200,
  });

  // Respond with the stream
  const stream = OpenAIStream(events, {
    onStart: async () => {
      // This callback is called when the stream starts
      // You can use this to save the prompt to your database
    },
    onToken: async (token: string) => {
      // This callback is called for each token in the stream
      // You can use this to debug the stream or save the tokens to your database
      // console.log("token " + token);
    },
    onCompletion: async (completion: string) => {
      // This callback is called when the stream completes
      // You can use this to save the final completion to your database
      // console.log("completion " + completion);
    },
  });

  let res = new StreamingTextResponse(stream);
  return res;
}
```

**Comment by sonphnt** [comment]
I have the same issue. Have you found a solution for this?

**Comment by IsuruCreative** [comment]
> I have the same issue. Have you found a solution for this?

No I haven't still.

**Comment by sonphnt** [comment]
Just found it. We could do like this.

`const stream = OpenAIStream(
    events as unknown as Stream<OpenAI.Chat.Completions.ChatCompletionChunk>, {
      async onCompletion(completion) {
        // your code here
      }
    }
  )

return new StreamingTextResponse(stream);`

**Comment by MaxLeiter** [comment]
Fixed in #912 and will be out in the next release.

**Comment by kshitijdomadia** [comment]
> Just found it. We could do like this.
> 
> ```
> const stream = OpenAIStream(
>     events as unknown as Stream<OpenAI.Chat.Completions.ChatCompletionChunk>, {
>       async onCompletion(completion) {
>         // your code here
>       }
>     }
>   )
> 
> return new StreamingTextResponse(stream);
> ```

This is the only solution that works and I am surprised that despite being fixed in the newer version it still doesn't work.

@MaxLeiter could this be because I am using azure_search for my use-case? apart from that there's nothing else being used

**Comment by lgrammel** [comment]
@kshitijdomadia the approach that you took has been deprecated in the new ai sdk. please check out https://sdk.vercel.ai/providers/ai-sdk-providers/azure for the new azure provider

**Comment by kshitijdomadia** [comment]
> @kshitijdomadia the approach that you took has been deprecated in the new ai sdk. please check out https://sdk.vercel.ai/providers/ai-sdk-providers/azure for the new azure provider

Thanks @lgrammel. I use AzureSearch as well. The new provider doesn't seem to have any documentation on it.

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Bug**  
Confidence: **High**

### Rationale
The issue reports concrete broken streaming behavior in the existing Azure OpenAI integration, and a maintainer confirmed that a code fix was made.

### Evidence
- Azure OpenAI responses passed through `OpenAIStream` and `StreamingTextResponse` arrived as one complete block instead of streaming incrementally.
- Another user reproduced the same behavior.
- Maintainer MaxLeiter stated that the issue was fixed in #912 and would be included in the next release.
